File: Scripts/results/generated_code_results/Flutter/bak_code/Education/EnglishLearningAppUI/SpeakScreen.dart
import 'package:flutter/material.dart';
class SpeakScreen extends StatefulWidget {
const SpeakScreen({Key? key}) : super(key: key);
@override
_SpeakScreenState createState() => _SpeakScreenState();
}
class _SpeakScreenState extends State<SpeakScreen> {
int _selectedIndex = 2;
String _selectedCategory = 'All';
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Speaking Course'),
),
body: SingleChildScrollView(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Padding(
padding: const EdgeInsets.all(16.0),
child: Text(
'Improve your speak skill with many courses',
style: TextStyle(fontSize: 16, color: Colors.grey),
),
),
Padding(
padding: const EdgeInsets.all(16.0),
child: Text(
'Studying Course',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
),
_buildFeaturedCourse(),
Padding(
padding: const EdgeInsets.all(16.0),
child: Text(
'Courses',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
),
SingleChildScrollView(
scrollDirection: Axis.horizontal,
padding: EdgeInsets.symmetric(horizontal: 16.0),
child: Row(
children: [
_buildCategoryButton('All'),
SizedBox(width: 8),
_buildCategoryButton('Beginner'),
SizedBox(width: 8),
_buildCategoryButton('Intermediate'),
SizedBox(width: 8),
_buildCategoryButton('Advanced'),
],
),
),
SizedBox(height: 16),
_buildCourseItem(
'Basic communication English',
'Primary',
'12 Lessons',
),
SizedBox(height: 16),
_buildCourseItem(
'Essential Structures',
'Primary',
'6 Lessons',
),
],
),
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: _selectedIndex,
onTap: (index) {
setState(() {
_selectedIndex = index;
});
if (index == 0) {
Navigator.pushNamed(context, 'Home');
} else if (index == 1) {
Navigator.pushNamed(context, 'Search');
}
},
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(icon: Icon(Icons.search), label: 'Search'),
BottomNavigationBarItem(icon: Icon(Icons.mic), label: 'Speak'),
BottomNavigationBarItem(icon: Icon(Icons.book), label: 'Vocabulary'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),
],
),
);
}
Widget _buildFeaturedCourse() {
return Container(
margin: EdgeInsets.all(16),
decoration: BoxDecoration(
borderRadius: BorderRadius.circular(16),
color: Colors.grey[200],
),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
ClipRRect(
borderRadius: BorderRadius.vertical(top: Radius.circular(16)),
child: Image.asset(
'assets/images/template.png',
height: 200,
width: double.infinity,
fit: BoxFit.cover,
),
),
Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Text(
'English REAL TALK',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
Container(
padding: EdgeInsets.symmetric(horizontal: 8, vertical: 4),
decoration: BoxDecoration(
color: Colors.green,
borderRadius: BorderRadius.circular(16),
),
child: Text(
'Intermediate',
style: TextStyle(color: Colors.white),
),
),
],
),
SizedBox(height: 8),
Text(
"Is speak English as easy but rigid as a textbook? No no! You're wrong!",
maxLines: 2,
overflow: TextOverflow.ellipsis,
),
SizedBox(height: 8),
ElevatedButton(
onPressed: () {
Navigator.pushNamed(context, 'Conversation');
},
child: Text('Start Course'),
),
],
),
),
],
),
);
}
Widget _buildCategoryButton(String category) {
return ElevatedButton(
onPressed: () {
setState(() {
_selectedCategory = category;
});
},
style: ElevatedButton.styleFrom(
backgroundColor: _selectedCategory == category ? Colors.blue : Colors.grey[300],
foregroundColor: _selectedCategory == category ? Colors.white : Colors.black,
),
child: Text(category),
);
}
Widget _buildCourseItem(String title, String level, String lessons) {
return Container(
margin: EdgeInsets.symmetric(horizontal: 16),
decoration: BoxDecoration(
borderRadius: BorderRadius.circular(16),
color: Colors.grey[200],
),
child: Row(
children: [
ClipRRect(
borderRadius: BorderRadius.horizontal(left: Radius.circular(16)),
child: Image.asset(
'assets/images/template.png',
width: 100,
height: 100,
fit: BoxFit.cover,
),
),
Expanded(
child: Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
title,
style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
),
SizedBox(height: 4),
Row(
children: [
Container(
padding: EdgeInsets.symmetric(horizontal: 8, vertical: 4),
decoration: BoxDecoration(
color: Colors.orange,
borderRadius: BorderRadius.circular(16),
),
child: Text(
level,
style: TextStyle(color: Colors.white, fontSize: 12),
),
),
SizedBox(width: 8),
Text(lessons, style: TextStyle(color: Colors.grey)),
],
),
],
),
),
),
IconButton(
icon: Icon(Icons.arrow_forward, color: Colors.blue),
onPressed: () {
Navigator.pushNamed(context, 'Conversation');
},
),
],
),
);
}
}

File: Scripts/results/generated_code_results/ablation_study/sam/Education/EnglishLearningAppUI/SpeakScreen.js
import React from 'react';

import { View, Text, Image, StyleSheet, ScrollView, TouchableOpacity } from 'react-native';

import Icon from 'react-native-vector-icons/Ionicons';

import { useNavigation } from '@react-navigation/native';



const SpeakScreen = () => {

  const navigation = useNavigation();



  const handleCoursePress = () => {

    navigation.navigate('Conversation');

  };



  return (

    <View style={styles.container}>

      <View style={styles.header}>

        <Text style={styles.headerTitle}>Speaking Course</Text>

        <Text style={styles.headerSubtitle}>Improve your speak skill with many courses</Text>

      </View>



      <ScrollView style={styles.content}>

        <View style={styles.featuredCourse}>

          <Image source={require('../assets/snack-icon.png')} style={styles.featuredImage} />

          <View style={styles.featuredInfo}>

            <View style={styles.levelTag}>

              <Text style={styles.levelText}>Intermediate</Text>

            </View>

            <Text style={styles.featuredTitle}>English REAL TALK</Text>

            <Text style={styles.featuredDescription}>

              Is speak English as easy but rigid as a textbook? No no! You're wrong! ...

            </Text>

            <TouchableOpacity style={styles.arrowButton}>

              <Icon name="arrow-forward" size={24} color="#000" />

            </TouchableOpacity>

          </View>

        </View>



        <View style={styles.courseTabs}>

          <ScrollView horizontal showsHorizontalScrollIndicator={false}>

            {['All', 'Beginner', 'Intermediate', 'Advanced'].map((tab, index) => (

              <TouchableOpacity key={index} style={[styles.tab, index === 0 && styles.activeTab]}>

                <Text style={[styles.tabText, index === 0 && styles.activeTabText]}>{tab}</Text>

              </TouchableOpacity>

            ))}

          </ScrollView>

        </View>



        <Text style={styles.sectionTitle}>Courses</Text>



        {[

          { title: 'Basic communication English', level: 'Primary', lessons: 12 },

          { title: 'Essential Structures', level: 'Primary', lessons: 6 },

        ].map((course, index) => (

          <TouchableOpacity key={index} style={styles.courseItem} onPress={handleCoursePress}>

            <Image source={require('../assets/snack-icon.png')} style={styles.courseImage} />

            <View style={styles.courseInfo}>

              <Text style={styles.courseTitle}>{course.title}</Text>

              <View style={styles.courseDetails}>

                <View style={styles.levelTag}>

                  <Text style={styles.levelText}>{course.level}</Text>

                </View>

                <Text style={styles.lessonCount}>{course.lessons} Lessons</Text>

              </View>

            </View>

            <TouchableOpacity style={styles.arrowButton}>

              <Icon name="arrow-forward" size={24} color="#000" />

            </TouchableOpacity>

          </TouchableOpacity>

        ))}

      </ScrollView>



      <View style={styles.bottomNav}>

        {['Home', 'Search', 'Speak', 'Vocabulary', 'Profile'].map((item, index) => (

          <TouchableOpacity

            key={index}

            style={styles.navItem}

            onPress={() => navigation.navigate(item)}

          >

            <Icon

              name={

                item === 'Home'

                  ? 'home'

                  : item === 'Search'

                  ? 'search'

                  : item === 'Speak'

                  ? 'mic'

                  : item === 'Vocabulary'

                  ? 'book'

                  : 'person'

              }

              size={24}

              color={item === 'Speak' ? '#007AFF' : '#888'}

            />

            <Text style={[styles.navText, item === 'Speak' && styles.activeNavText]}>{item}</Text>

          </TouchableOpacity>

        ))}

      </View>

    </View>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#F5F5F5',

  },

  header: {

    padding: 20,

    backgroundColor: '#FFFFFF',

  },

  headerTitle: {

    fontSize: 24,

    fontWeight: 'bold',

  },

  headerSubtitle: {

    fontSize: 14,

    color: '#666',

    marginTop: 5,

  },

  content: {

    flex: 1,

  },

  featuredCourse: {

    backgroundColor: '#FFFFFF',

    margin: 15,

    borderRadius: 10,

    overflow: 'hidden',

  },

  featuredImage: {

    width: '100%',

    height: 150,

  },

  featuredInfo: {

    padding: 15,

  },

  levelTag: {

    backgroundColor: '#E8F5E9',

    paddingHorizontal: 10,

    paddingVertical: 5,

    borderRadius: 15,

    alignSelf: 'flex-start',

  },

  levelText: {

    color: '#4CAF50',

    fontSize: 12,

    fontWeight: 'bold',

  },

  featuredTitle: {

    fontSize: 18,

    fontWeight: 'bold',

    marginTop: 10,

  },

  featuredDescription: {

    fontSize: 14,

    color: '#666',

    marginTop: 5,

  },

  arrowButton: {

    position: 'absolute',

    right: 15,

    bottom: 15,

  },

  courseTabs: {

    flexDirection: 'row',

    paddingHorizontal: 15,

    marginBottom: 15,

  },

  tab: {

    paddingHorizontal: 20,

    paddingVertical: 10,

    marginRight: 10,

    borderRadius: 20,

    backgroundColor: '#FFFFFF',

  },

  activeTab: {

    backgroundColor: '#007AFF',

  },

  tabText: {

    color: '#000',

  },

  activeTabText: {

    color: '#FFFFFF',

  },

  sectionTitle: {

    fontSize: 18,

    fontWeight: 'bold',

    marginLeft: 15,

    marginBottom: 10,

  },

  courseItem: {

    flexDirection: 'row',

    backgroundColor: '#FFFFFF',

    marginHorizontal: 15,

    marginBottom: 15,

    borderRadius: 10,

    overflow: 'hidden',

  },

  courseImage: {

    width: 80,

    height: 80,

  },

  courseInfo: {

    flex: 1,

    padding: 15,

  },

  courseTitle: {

    fontSize: 16,

    fontWeight: 'bold',

  },

  courseDetails: {

    flexDirection: 'row',

    alignItems: 'center',

    marginTop: 5,

  },

  lessonCount: {

    fontSize: 12,

    color: '#666',

    marginLeft: 10,

  },

  bottomNav: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    alignItems: 'center',

    backgroundColor: '#FFFFFF',

    paddingVertical: 10,

    borderTopWidth: 1,

    borderTopColor: '#E0E0E0',

  },

  navItem: {

    alignItems: 'center',

  },

  navText: {

    fontSize: 12,

    color: '#888',

    marginTop: 5,

  },

  activeNavText: {

    color: '#007AFF',

  },

});



export default SpeakScreen;